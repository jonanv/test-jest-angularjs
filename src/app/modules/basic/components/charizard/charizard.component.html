<h1>Pok√©mon</h1>

<ng-container *ngIf="!pokemon; else elseTemplate">
  <h2>Loading...</h2>
</ng-container>
<ng-template #elseTemplate>
  <h3>{{ pokemon?.name | titlecase }}</h3>
  <img
    [src]="pokemon?.sprites?.front_default"
    [alt]="pokemon?.name">
</ng-template>
