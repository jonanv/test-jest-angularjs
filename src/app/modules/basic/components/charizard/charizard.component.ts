import { Component, OnInit } from '@angular/core';

// Imports
import { first } from 'rxjs';

// Services
import { PokemonService } from '../../services/pokemon.service';

// Interfaces
import { Pokemon } from '../../interfaces/pokemon.interface';

@Component({
  selector: 'app-charizard',
  templateUrl: './charizard.component.html',
  styles: ``
})
export class CharizardComponent implements OnInit {

  public pokemon!: Pokemon | null;

  private loading: boolean = false;
  private error: any;

  constructor(
    private pokemonService: PokemonService
  ) { }

  public ngOnInit(): void {
    this.getPokemon();
  }

  public getPokemon(): void {
    this.pokemonService.getPokemon(6)
      .pipe(first())
      .subscribe({
        next: (response: Pokemon) => {
          this.pokemon = response;
          console.log(this.pokemon);
          this.loading = false;
        },
        error: (error: any) => {
          this.error = error.error;
          // console.log(this.error);
          this.loading = false;
        },
        complete: () => {
          // console.log('complete');
        }
      });
  }

}
