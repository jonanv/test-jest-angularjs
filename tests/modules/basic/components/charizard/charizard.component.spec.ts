import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CharizardComponent } from '../../../../../src/app/modules/basic/components/charizard/charizard.component';

describe('CharizardComponent', () => {
  let component: CharizardComponent;
  let fixture: ComponentFixture<CharizardComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [CharizardComponent]
    })
    .compileComponents();

    fixture = TestBed.createComponent(CharizardComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  test('Debe crear el componente charizard', () => {
    expect(component).toBeTruthy();
  });

});
