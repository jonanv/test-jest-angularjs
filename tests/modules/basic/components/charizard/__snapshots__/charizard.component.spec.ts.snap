// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharizardComponent Debe de hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Pok√©mon
  </h1>
  <h2>
    Loading...
  </h2>
  
  
  
</div>
`;
